### PR TITLE
[FLINK-29219][table]fix CREATE TABLE AS statement blocks SQL client's execution

### PR DESCRIPTION
**Pull request by Tartarus0zm**
## What is the purpose of the change

* fix CTAS statement blocks SQL client's execution

## Brief change log

`ContextResolvedTable` objects are constructed manually and not obtained using `CatalogManager`.

## Verifying this change

 * The change was verified manually by using sql-client.sh to execute the CTAS statement with `table.dml-sync=false` then the sql client was not blocked.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/CreateTableASOperation.java [hunk 1]
```diff
@@ -38,25 +38,28 @@ public class CreateTableASOperation implements CreateOperation {
...
         this.sinkModifyOverwrite = sinkModifyOverwrite;
+        this.contextResolvedTable = contextResolvedTable;
     }
 
     public CreateTableOperation getCreateTableOperation() {
         return createTableOperation;
     }
 
-    public SinkModifyOperation toSinkModifyOperation(CatalogManager catalogManager) {
+    public SinkModifyOperation toSinkModifyOperation() {
```
Comment (lsyldliu) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/CreateTableASOperation.java:60]: Please revert the changes in this class, it is no need now.

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlCreateTableConverter.java [hunk 2]
```diff
@@ -121,8 +121,13 @@ Operation convertCreateTableAS(
                         sqlCreateTableAs.isIfNotExists(),
                         sqlCreateTableAs.isTemporary());
 
+        ContextResolvedTable contextResolvedTable =
```
Comment (lsyldliu) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlCreateTableConverter.java:124]: Ditto

**Comment by lsyldliu** [comment]
Let's continue this work in https://github.com/apache/flink/pull/20869.

**Comment by lsyldliu** [comment]
@Tartarus0zm close this pr?

**Comment by Tartarus0zm** [comment]
@lsyldliu  ok, thanks for your work
